components/match2/wikis/clashroyale/match_group_display_bracket_custom.lua

```diff
@@ -0,0 +1,43 @@
+---
+-- @Liquipedia
+-- wiki=clashroyale
+-- page=Module:MatchGroup/Display/Bracket/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local BracketDisplay = Lua.import('Module:MatchGroup/Display/Bracket', {requireDevIfEnabled = true})
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+
+local CustomBracketDisplay = Table.copy(BracketDisplay)
+
+function CustomBracketDisplay.BracketContainer(props)
+	local bracket = MatchGroupUtil.fetchMatchGroup(props.bracketId)
+	return BracketDisplay.Bracket({
+		bracket = bracket,
+		config = Table.merge(props.config, {
+			opponentHeight = CustomBracketDisplay.computeBracketOpponentHeight(bracket.matchesById),
+		})
+	})
+end
+
+local defaultOpponentHeights = {
+	default = 17 + 6,
+	duo = 2 * 17 + 6 + 4,
+}
+function CustomBracketDisplay.computeBracketOpponentHeight(matchesById)
+	local maxHeight = defaultOpponentHeights.default
+
+	for _, match in pairs(matchesById) do
+		for _, opponent in ipairs(match.opponents) do
+			maxHeight = math.max(maxHeight, defaultOpponentHeights[opponent.type] or 0)
+		end
+	end
+
+	return maxHeight
+end
```

Comment:
```suggestion
local defaultOpponentHeights = {
	duo = 2 * 17 + 6 + 4,
}
function CustomBracketDisplay.computeBracketOpponentHeight(matchesById)
	local maxHeight = 0

	for _, match in pairs(matchesById) do
		for _, opponent in ipairs(match.opponents) do
			maxHeight = math.max(maxHeight, defaultOpponentHeights[opponent.type] or 0)
		end
	end

	return maxHeight
end
```
To not have hardcoded default height

Reply:
that actually makes the height 0
would have to adjust the other function to ignore 0
![Screenshot 2023-01-10 13 57 23](https://user-images.githubusercontent.com/75081997/211557760-814d5dbd-b737-4740-87fb-a02b7829ce5a.png)

Reply:
https://liquipedia.net/clashroyale/index.php?title=Module%3AMatchGroup%2FDisplay%2FBracket%2FCustom&type=revision&diff=122039&oldid=120370